src/ray/core_worker/test/thread_pool_test.cc

```diff
@@ -0,0 +1,61 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ray/core_worker/transport/thread_pool.h"
+
+#include <gtest/gtest.h>
+
+#include <atomic>
+#include <future>
+
+namespace ray {
+namespace core {
+
+TEST(BoundedExecutorTest, InitializeThreadCallbackAndReleaserAreCalled) {
```

Comment:
Awesome! 🙌

Let's also add some basic test cases for `.Post()`? For example, post N work items for N threads, check that all are in progress, post N+1'th item and check it blocks until one thread becomes unblocked.

Reply:
Added https://github.com/ray-project/ray/pull/52575/commits/8123c329deecc64e779b348f03759bf06be77277